File: js/task2/S2_ARR_06.js
function getActiveUserNames(users) {
  const activeUsers = users.filter(user => user.active);
  const activeNames = activeUsers.map(user => user.name.toUpperCase());
  return activeNames.sort();
}

const users = [
  { id: 1, name: "alice", active: true },
  { id: 2, name: "bob", active: false },
  { id: 3, name: "charlie", active: true },
  { id: 4, name: "David", active: true }
];

console.log(getActiveUserNames(users));
console.log(getActiveUserNames([]));

File: python/task2/S2_ARR_06.py
def getActiveUserNames(users):
  active_users = [user for user in users if user.get("active")]
  active_names = [user["name"].upper() for user in active_users]
  return sorted(active_names)

users = [
  {"id": 1, "name": "alice", "active": True},
  {"id": 2, "name": "bob", "active": False},
  {"id": 3, "name": "charlie", "active": True},
  {"id": 4, "name": "David", "active": True}
]

print(getActiveUserNames(users))
print(getActiveUserNames([]))
